rpcs3/Emu/RSX/RSXFIFO.cpp

```diff
@@ -556,6 +556,14 @@ namespace rsx
 			if (auto method = methods[reg])
 			{
 				method(this, reg, value);
+
+				if (invalid_command_interrupt_raised && g_cfg.video.FIFO_corruption_detection)
```

Comment:
Will it really help though? I mean, even now, invalid methods don't really do much other than spam the log and the hangs usually arise from other side-effects (bad jumps, missed semaphore releases, etc).

Reply:
Those commands usually pass on some *valid* methods as they tend to have extremely large counts, modifiying them and exexecute them with possibly corrupted contents.
Thats why theres an "invalid location" exception in the log of the mentioned issue, even when the same command is still executing, (corrupted semaphore location then executed a semaphore method)

Reply:
```·E 0:00:38.935238 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x24 (arg=0x80a78)
·E 0:00:38.935240 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x28 (arg=0x3fc00000)
·E 0:00:38.935242 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x2c (arg=0x40a00000)
·E 0:00:38.935244 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x30 (arg=0x40a74)
·E 0:00:38.935246 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x34 (arg=0x1)
·E 0:00:38.935247 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x38 (arg=0x40a70)
·E 0:00:38.935249 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x3c (arg=0x1)
·E 0:00:38.935251 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x40 (arg=0x40a6c)
·E 0:00:38.935252 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x44 (arg=0x203)
·E 0:00:38.935254 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x48 (arg=0x40310)
·E 0:00:38.935255 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x4c (arg=0x0)
·E 0:00:38.935257 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x54 (arg=0x0)
·E 0:00:38.935259 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x58 (arg=0x40324)
·E 0:00:38.935261 {RSX [0xf7033a4]} RSX: Invalid RSX method 0x5c (arg=0x0)
·F 0:00:38.935377 {RSX [0xf7033a4]} RSX: class std::runtime_error thrown: Invalid location (offset=0x405, location=0x41830)```
notice how GET address didnt change, indicating it is actually the same command.

Reply:
Won't it still crash after this bogus method chain is executed?

Reply:
Since there are much more invalid methods with bigr gaps, than valid ones with small range. theres a good chance no important register has been corrupted.
Its likely to execute some bad jumpor invalid cmd and reset the queue afterwards, Im not enforcing it becauase of the chance that the methods are actually valid and cause an infinite loop.

Reply:
My point is that if it doesn't work in practice then there is no point in having the extra code. If it actually fixes desyncs in any game then its fine by me.

Reply:
It used to handle such invalid methods fine in the past and making games progress further,
Now its usually one of the last things you will find in logs because of how its handled now.

Reply:
For ease of merging Ive just reduced it to the segfaults fix, feel free to add it later on if you want to.